cql2pgjson-cli/src/main/java/org/z3950/zing/cql/cql2pgjsoncli/CQL2PGCLIMain.java

```diff
@@ -0,0 +1,142 @@
+        .hasArg()
+        .desc("A JSON string or a pathname to a JSON file to describe the db/schema map")
+        .build();
+
+    Option dbschema = Option.builder("b")
+        .hasArg()
+        .desc("Path to RMB-style schema.json to describe database")
+        .build();
+
+    options.addOption(schema);
+    options.addOption(database);
+    options.addOption(field);
+    options.addOption(schemamap);
+    options.addOption(dbschema);
+
+    CommandLineParser parser = new DefaultParser();
+    CommandLine line = parser.parse(options, args);
+    CQL2PgJSON cql2pgJson = null;
+    String fullFieldName = line.getOptionValue("t") + "." + line.getOptionValue("f", "json");
+    if(!line.hasOption("m")) {
+      if(line.hasOption("b")) {
+        cql2pgJson = new CQL2PgJSON(fullFieldName, null, line.getOptionValue("b"));
+      } else if(line.hasOption("s")) {
+        String schemaText = readFile(line.getOptionValue("s"), Charset.forName("UTF-8"));
+        cql2pgJson = new CQL2PgJSON(fullFieldName, schemaText);
+      } else {
+        cql2pgJson = new CQL2PgJSON(fullFieldName); //No schemas
+      }
+    } else {
+      Map<String, String> fieldSchemaMap = parseDatabaseSchemaString(line.getOptionValue("m"));
+      cql2pgJson = new CQL2PgJSON(fieldSchemaMap);
+    }
+    List<String> cliArgs = line.getArgList();
+    String cql = cliArgs.get(0);
+    return parseCQL(cql2pgJson, line.getOptionValue("t"), cql);
+  }
+
+  static String readFile(String path, Charset encoding) throws IOException
+  {
+    System.out.println("Reading file " + path);
+    byte[] encoded = Files.readAllBytes(Paths.get(path));
+    String content = new String(encoded, encoding);
+    //System.out.println(String.format("Content of %s is: %s", path, content));
+    return content;
+  }
+
+  static protected String parseCQL(CQL2PgJSON cql2pgJson, String dbName, String cql) throws IOException,
+      FieldException, SchemaException, QueryValidationException {
+    SqlSelect sql = cql2pgJson.toSql(cql);
+    return String.format("select * from %s where %s order by %s",
+        dbName, sql.getWhere(), sql.getOrderBy());
+  }
+
+  /*
+    If the string is valid JSON, read the values from the JSON object. If the
+    string is a path to a JSON file, load the file and read the JSON from the
+    file
+  */
+  static Map<String, String> parseDatabaseSchemaString(String dbsString) throws
+      IOException {
+    JSONObject fieldSchemaJson = null;
+    Map<String, String> fieldSchemaMap = new HashMap<>();
+    try {
+      fieldSchemaJson = new JSONObject(dbsString);
+    } catch( JSONException je ) {
+      System.out.println(String.format("Unable to parse %s as JSON: %s",
+          dbsString, je.getLocalizedMessage()));
+    }
+    if(fieldSchemaJson == null) {
+      String fieldSchemaJsonText = readFile(dbsString, Charset.forName("UTF-8"));
+      fieldSchemaJson = new JSONObject(fieldSchemaJsonText);
+    }
+    if(fieldSchemaJson == null) {
```

Comment:
This can never be null because `new JSONObject` is never null.
Can you remove this if clause?

Reply:
Removed.